src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          state[key] = action.extraData[key];
+        }
+        break;
+      case 'removeProperties':
+        for (const keyToRemove in state) {
```

Comment:
Instead of looping through all keys in the state, it's more efficient to loop through 'keysToRemove' and check if the key exists in the state.